evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
```

Comment:
I've just moved that to the `get_vm()` method as that way `get_vm_class_for_block_number()` returns the value passed in `vm_configuration` without modifying it, which makes testing easier.

But I just noticed that although `get_vm_class_for_block_number()` is a classmethod, it is only called from `get_vm()`, which is a regular method. Do you expect there to be other places that will call `get_vm_class_for_block_number()` using a EVM class or should it be a regular method as well?